### ISSUE TITLE
Invalid checksum for version 6.5.4

### ISSUE THREAD

**Issue by anomiex** [roots/wordpress-no-content#7]
### Version

6.5.4

### What did you expect to happen?

```console
$ composer require roots/wordpress-no-content
./composer.json has been created
Running composer update roots/wordpress-no-content
Loading composer repositories with package information
Updating dependencies
Lock file operations: 1 install, 0 updates, 0 removals
  - Locking roots/wordpress-no-content (6.5.4)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 1 install, 0 updates, 0 removals
  - Installing roots/wordpress-no-content (6.5.4): Extracting archive
1 package suggestions were added by new dependencies, use `composer suggest` to see details.
Generating autoload files
1 package you are using is looking for funding.
Use the `composer fund` command to find out more!
No security vulnerability advisories found.
Using version ^6.5 for roots/wordpress-no-content
```
```console
$ du -shx vendor/roots/wordpress-no-content/
63M	vendor/roots/wordpress-no-content/
```

### What actually happens?

```console
$ composer require roots/wordpress-no-content
./composer.json has been created
Running composer update roots/wordpress-no-content
Loading composer repositories with package information
Updating dependencies
Lock file operations: 1 install, 0 updates, 0 removals
  - Locking roots/wordpress-no-content (6.5.4)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 1 install, 0 updates, 0 removals
  - Downloading roots/wordpress-no-content (6.5.4)
    Failed to download roots/wordpress-no-content from dist: The checksum verification of the file failed (downloaded from https://downloads.wordpress.org/release/wordpress-6.5.4-no-content.zip)
    Now trying to download from source
  - Syncing roots/wordpress-no-content (6.5.4) into cache
  - Installing roots/wordpress-no-content (6.5.4): Cloning 6.5.4 from cache
1 package suggestions were added by new dependencies, use `composer suggest` to see details.
Generating autoload files
1 package you are using is looking for funding.
Use the `composer fund` command to find out more!
No security vulnerability advisories found.
Using version ^6.5 for roots/wordpress-no-content
```
```console
$ du -shx vendor/roots/wordpress-no-content/
597M	vendor/roots/wordpress-no-content/
```

### Steps to reproduce

1. Run `composer require roots/wordpress-no-content` in an empty project.

### Please confirm this isn't a support request.

Yes

**Comment by alexandrebuffet** [comment]
Same problem here!

**Comment by LeoColomb** [comment]
I confirm the issue for both `wordpress-no-content` and `wordpress-full`, only on version 6.5.4.

* `wordpress-no-content`:
  * [Stored shasum](https://github.com/roots/wordpress-no-content/commit/c111987d5f464bd3a70e31aef978bf903d9c4c4c#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34R12): `d6b3a78653efeff8edd463b2c24d62228bb98284`
  * [Current shasum](https://downloads.wordpress.org/release/wordpress-6.5.4-no-content.zip.sha1): `bb4d48b2f6f47f4c06061460c83f8453e9809af5`

* `wordpress-full`:
  * [Stored shasum](https://github.com/roots/wordpress-full/commit/00fd39122a927421cfc2dc039e2f067bf6e328f9#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34R12): `db7aff65252b5c005688daa156baf11fc7f8cbb7`
  * [Current shasum](https://downloads.wordpress.org/release/wordpress-6.5.4.zip.sha1): `342c038cfee1e8eef7d64cfc0207af7cc351034c`

It's near impossible to find the history of this change nor information regarding this event.
The sum is [gathered from WP.org API](https://github.com/roots/wordpress-packager/blob/ecba81bcf852101d74d323aa0d88af94c441d464/src/ReleaseSources/WPDotOrgAPI.php#L59-L62), not generated locally.

I'd suggest deleting the tags and let the packager recreate them with the proper sum.
@retlehs @swalkinshaw Looks good?

**Comment by LeoColomb** [comment]
@anomiex If you're still working at Automattic, would it be possible to request an investigation? (edit: if that makes sense with WP.org, not sure actually.)

**Comment by Martin-Welte** [comment]
Have the same issue - Version 6.5.3 Is ok though. @LeoColomb i do get the same checksum as you.

**Comment by swalkinshaw** [comment]
@LeoColomb yeah re-creating them is our only solution looks like 👍

**Comment by LeoColomb** [comment]
Should be resolved now.

**Comment by tbradsha** [comment]
Perhaps it's related to this:
https://core.trac.wordpress.org/changeset/58345

**Comment by jGRUBBS** [comment]
I'm still experiencing this issue when I try to `composer update` even when I add `--no-cache`
![Screenshot 2024-06-06 at 8 58 38 AM](https://github.com/roots/wordpress-no-content/assets/338605/726033e3-9c10-48ae-8e16-198416bb46de)
Am I missing something?

**Update:**
This is working for me as of 9:20am EDT

**Comment by anomiex** [comment]
> @anomiex If you're still working at Automattic, would it be possible to request an investigation? (edit: if that makes sense with WP.org, not sure actually.)

My working at Automattic doesn't give me any special ability to request an investigation compared to anyone else who isn't already involved in the WordPress.org community. 🙂

If it would have any effect at all, my working at Automattic might be a detriment if some see it as "Automattic" trying to throw their weight around or something. While I'd disagree with that, I can understand where they might be coming from in being wary.

**Comment by dd32** [comment]
Hi folks - I got a ping for this thread :)

Looking at the logs, it appears that the 6.5.4 ZIPs were rebuilt ~4hrs after release, that accounts for why the SHA1 changed. 

I have verified the contents of the ZIPs are the same, as I have access to both - there are some differences due to timestamps between the files, which accounts for the hash change.

While we'd prefer zips not be rebuilt, it's not a guarantee that it won't happen, and during the release time it's somewhat common for ZIPs to be rebuilt. It's rare that it happens after release, but not guaranteed.

The SHA1 hashes provided are intended on being the hash of the ZIP file at the current time, not intended on being stored and compared later on, but I understand why you are.

That being said; I've filed a task to add some enhancements here to ensure that overwriting an existing ZIP is the intended operation; https://meta.trac.wordpress.org/ticket/7667 as there was no reason for the rebuild in this case.

**Comment by LeoColomb** [comment]
Thanks a lot for your comment, @dd32, appreciated.
I'm a bit of surprise that the checksum is not expected to be stable _after_ release, though.
How do you usually assess the stability of the artifacts (e.g, not been altered and summed by an intruder)?

**Comment by tbradsha** [comment]
Personally I think it's a reasonable expectation that a release zip checksum should be stable; a definitive release is "written in stone", so to speak, and shouldn't change. If it changes without a version bump, that raises all sorts of red flags with regards to security: it's clear the zip changed, but there's no way to know _what_ changed (since to my knowledge, the first zip doesn't exist for comparison).

This is perhaps even more relevant when distributing via `composer`, as updates are often invoked via automations that few will see.

**Comment by m** [comment]
If the ZIPs were regenerated with the exact same content inside, why is the hash different?

**Comment by tbradsha** [comment]
> If the ZIPs were regenerated with the exact same content inside, why is the hash different?

If comparing individual (unzipped) files, checksums would remain the same, but zipped files would contain all the file metadata (e.g. timestamps, which [the above comment](https://github.com/roots/wordpress-no-content/issues/7#issuecomment-2153744037) indicates was the difference).

Probably the best place for discussing this is [in the meta Trac ticket](https://meta.trac.wordpress.org/ticket/7667), though, as this specific repo/package is working as expected.

**Comment by dd32** [comment]
Just noting that https://meta.trac.wordpress.org/ticket/7667 has been resolved.

> If the ZIPs were regenerated with the exact same content inside, why is the hash different?

For some history on this; neither ZIP nor TAR are deterministic compression formats. Although they can be forced to be.
 - SVN export sets the file timestamps to the last-commit date, but directories are created with the current timestamp.
 - ZIPs store modification dates, including of directories (see above).
 - ZIPs store additional metadata about the file, which are not needed and could be excluded.
 - TARs `gnu` format can change between versions of TAR, although usually doesn't.
 - TARs include additional Metadata about the files, such as the UserID / GroupID / UserName / GroupName of the creator, and Access+Creation in addition to Modification timestamps.
 - TARs when compressed with gzip via `-z` include the Tar ProcessID and current timestamp in the GZIP Checksums.

Changes to the packaging UI should reduce any potential accidental rebuilds, and I've applied some changes to the packaging scripts to force the `.zip` and `.tar.gz`'s to be more expected as above.
